cmd/golangci-lint/main.go

```diff
@@ -13,31 +13,64 @@ var (
 	goVersion = "unknown"
 
 	// Populated by goreleaser during build
-	version = "master"
+	version = "unknown"
 	commit  = "?"
 	date    = ""
 )
 
 func main() {
-	if buildInfo, available := debug.ReadBuildInfo(); available {
-		goVersion = buildInfo.GoVersion
+	info := createBuildInfo()
 
-		if date == "" {
-			version = buildInfo.Main.Version
-			commit = fmt.Sprintf("(unknown, mod sum: %q)", buildInfo.Main.Sum)
-			date = "(unknown)"
-		}
+	if err := commands.Execute(info); err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "failed executing command with error %v\n", err)
+		os.Exit(exitcodes.Failure)
 	}
+}
 
+func createBuildInfo() commands.BuildInfo {
 	info := commands.BuildInfo{
-		GoVersion: goVersion,
-		Version:   version,
 		Commit:    commit,
+		Version:   version,
+		GoVersion: goVersion,
 		Date:      date,
 	}
 
-	if err := commands.Execute(info); err != nil {
-		_, _ = fmt.Fprintf(os.Stderr, "failed executing command with error %v\n", err)
-		os.Exit(exitcodes.Failure)
+	if buildInfo, available := debug.ReadBuildInfo(); available {
+		info.GoVersion = buildInfo.GoVersion
+
+		if date == "" {
+			info.Version = buildInfo.Main.Version
+
+			var revision string
+			var modified string
+			for _, setting := range buildInfo.Settings {
+				// The `vcs.xxx` information is only available with `go build`.
+				// This information is are not available with `go install` or `go run`.
```

Comment:
`is are` 🤔

Reply:
french mistakes: in French `information` can be a plural but not in English, so brain bug.